designer/server/src/main/scala/pl/touk/nussknacker/ui/statistics/FingerprintService.scala

```diff
@@ -16,26 +16,30 @@ class FingerprintService(dbioRunner: DBIOActionRunner, fingerprintRepository: Fi
     implicit ec: ExecutionContext
 ) extends LazyLogging {
 
-  def fingerprint(config: UsageStatisticsReportsConfig, fingerprintFileName: FileName): Future[Fingerprint] = {
+  def fingerprint(
+      config: UsageStatisticsReportsConfig,
+      fingerprintFileName: FileName
+  ): Future[Either[StatisticError, Fingerprint]] = {
     // We filter out blank fingerprint and source because when smb uses docker-compose, and forwards env variables eg. USAGE_REPORTS_FINGERPRINT
     // from system and the variable doesn't exist, there is no way to skip variable - it can be only set to empty
     config.fingerprint.filterNot(_.isBlank) match {
-      case Some(fingerprintFromConfig) => Future.successful(new Fingerprint(fingerprintFromConfig))
+      case Some(fingerprintFromConfig) => Future.successful(Right(new Fingerprint(fingerprintFromConfig)))
       case None                        => fetchOrGenerate(fingerprintFileName)
     }
   }
 
-  private def fetchOrGenerate(fingerprintFileName: FileName): Future[Fingerprint] =
+  private def fetchOrGenerate(fingerprintFileName: FileName): Future[Either[StatisticError, Fingerprint]] =
     for {
-      dbFingerprint <- dbioRunner.run(fingerprintRepository.read())
+      dbFingerprint <- dbioRunner.safeRun(fingerprintRepository.read(), _ => DbError)
```

Comment:
shouldn't we have a `read` method that returns an `Option[]`?

Reply:
we have and in the second parameter we provide the mapping function from exception to domain